compiler/ballerina-lang/src/main/java/io/ballerina/projects/NullBackend.java

```diff
@@ -0,0 +1,101 @@
+/* Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+*/
+package io.ballerina.projects;
+
+import io.ballerina.projects.environment.ProjectEnvironment;
+import io.ballerina.projects.internal.DefaultDiagnosticResult;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Ballerina compiler backend that does not emit backend code, used for testing BIR.
+ *
+ * @since 2.0
+ */
+public class NullBackend extends CompilerBackend {
+    private final PackageCompilation compilation;
+    private final PackageContext packageContext;
+    private final CompilerContext compilerContext;
+    private DefaultDiagnosticResult diagnosticResult;
+
+    public NullBackend(PackageCompilation compilation) {
+        this.compilation = compilation;
+        this.packageContext = compilation.packageContext();
+        ProjectEnvironment projectEnvContext = this.packageContext.project().projectEnvironmentContext();
+        this.compilerContext = projectEnvContext.getService(CompilerContext.class);
+    }
+
+    public void compile() {
```

Comment:
This is actually not compile, but generate BIR right? Shall we rename it accordingly? Also, should we have this method explicitly called? Why don't we move the call to this method to the constructor?

Reply:
Updated to use a static factory method similar to JBallerinaBackend. Please check.